src/Renderer/ThreeExtended/GlobeControls.js

```diff
@@ -689,17 +691,49 @@ function GlobeControls(camera, target, domElement, engine) {
         offset.copy(this.camera.position);
         offset.applyMatrix4(cameraTargetOnGlobe.matrixWorldInverse);
         spherical.setFromVector3(offset);
+
+        if (state === CONTROL_STATE.ORBIT && (Math.abs(snapShotSpherical.phi - spherical.phi) > delta || Math.abs(snapShotSpherical.theta - spherical.theta) > delta)) {
+            this.dispatchEvent({
+                type: 'orientation-changed',
+                previous: {
+                    tilt: snapShotSpherical.phi * 180 / Math.PI,
+                    heading: snapShotSpherical.theta * 180 / Math.PI,
+                },
+                new: {
+                    tilt: spherical.phi * 180 / Math.PI,
+                    heading: spherical.theta * 180 / Math.PI,
+                },
+            });
+        } else if (state === CONTROL_STATE.PAN) {
+            this.dispatchEvent({
+                type: 'pan-changed',
+            });
+        }
+
+        const previousRange = snapShotSpherical.radius;
+        const newRange = this.getRange();
+        if (Math.abs(newRange - previousRange) / previousRange > 0.001) {
+            this.dispatchEvent({
+                type: 'range-changed',
```

Comment:
this event is dispatched twice when range =< 10000 in `setCameraTargetGeoPosition`